core/baseline-core-v1.0-psd01.md

```diff
@@ -191,8 +191,273 @@ The ability of a Party operating Workflows on a baseline-compliant implementatio
 
 # 2 Design and Architecture
 
+This section provides definitions, key concepts, and overviews of the components of a Baseline Protocol Implementation compliant with the requirements of this document. 
+
+## 2.1 Agreement
+An Agreement between two or more Counterparties allows for transactions between them dealing with, for example, commercial items such as products. An Agreement governs and defines all transactions between counterparties.
```

Comment:
An agreement is a manifestation of mutual assent by two or more parties to one another. (https://www.law.cornell.edu/wex/agreement)  In the context of a baseline workflow, an agreement is the mutual assent of two or more parties to a set of rules governing the creation, modification and deletion of records held in the respective parties' information systems such that the records can be verified to be maintained in a state of consistency and that state changes can be verified to have been executed correctly in accordance with the Agreement's rules.  The terms of an Agreement may be written or unwritten. The rules governing the part of an Agreement that involves record-keeping state changes must be digitally executable by a system compliant with the Baseline Protocol in order to verify consistency and correctness.

Reply:
@humbitious is this a comment or a change request? Happy to include a version of this at the right places ... in this case n the contract and commercial document section. Happy to add the Cornell reference and wording in 2.1 -> do you have a stable reference (aka not only a URL)? Also, the section on BPI has a lot of the wording you included. I would stay away from baseline workflow etc until the relevant sections and focus on the primary function of mutually agreed upon state changes on state objects. Will push a new commit in a scond.

Reply:
I prefer the definition from @humbitious. Chapter 2.4 sounds too legal! I can't comment on that but it sounds we are widening the scope of Baseline to decide for a commercial legal contract which could be an instant road-blocker for its adoption. I expect Baseline's scope to be clearly on the technical aspects of reaching technical(digital) consistency among different systems of record.